pkg/sql/sem/tree/cast.go

```diff
@@ -768,201 +850,225 @@ var castMap = map[oid.Oid]map[oid.Oid]cast{
-		oid.T_timestamp:   {maxContext: CastContextImplicit, origin: contextOriginPgCast, volatility: volatilityTODO},
-		oid.T_timestamptz: {maxContext: CastContextImplicit, origin: contextOriginPgCast, volatility: volatilityTODO},
+		oid.T_date:        {maxContext: CastContextAssignment, origin: contextOriginPgCast, volatility: VolatilityImmutable},
+		oid.T_float4:      {maxContext: CastContextExplicit, origin: contextLegacyConversion, volatility: VolatilityImmutable},
+		oid.T_float8:      {maxContext: CastContextExplicit, origin: contextLegacyConversion, volatility: VolatilityImmutable},
+		oid.T_int2:        {maxContext: CastContextExplicit, origin: contextLegacyConversion, volatility: VolatilityImmutable},
+		oid.T_int4:        {maxContext: CastContextExplicit, origin: contextLegacyConversion, volatility: VolatilityImmutable},
+		oid.T_int8:        {maxContext: CastContextExplicit, origin: contextLegacyConversion, volatility: VolatilityImmutable},
+		oid.T_numeric:     {maxContext: CastContextExplicit, origin: contextLegacyConversion, volatility: VolatilityImmutable},
+		oid.T_time:        {maxContext: CastContextAssignment, origin: contextOriginPgCast, volatility: VolatilityImmutable},
+		oid.T_timestamp:   {maxContext: CastContextImplicit, origin: contextOriginPgCast, volatility: VolatilityImmutable},
+		oid.T_timestamptz: {maxContext: CastContextImplicit, origin: contextOriginPgCast, volatility: VolatilityStable},
 		// Automatic I/O conversions to string types.
 		oid.T_bpchar: {
 			maxContext: CastContextAssignment,
 			origin:     contextOriginAutomaticIOConversion,
-			volatility: volatilityTODO,
+			// TODO(mgartner): This should be stable.
+			volatility: VolatilityImmutable,
 			volatilityHint: "TIMESTAMP to CHAR casts are dependent on DateStyle; consider " +
 				"using to_char(timestamp) instead.",
 			dateStyleAffected: true,
 		},
 		oid.T_char: {
 			maxContext: CastContextAssignment,
 			origin:     contextOriginAutomaticIOConversion,
-			volatility: volatilityTODO,
+			// TODO(mgartner): This should be stable.
+			volatility: VolatilityImmutable,
 			volatilityHint: `TIMESTAMP to "char" casts are dependent on DateStyle; consider ` +
 				"using to_char(timestamp) instead.",
 			dateStyleAffected: true,
 		},
 		oid.T_name: {
 			maxContext: CastContextAssignment,
 			origin:     contextOriginAutomaticIOConversion,
-			volatility: volatilityTODO,
+			// TODO(mgartner): This should be stable.
+			volatility: VolatilityImmutable,
 			volatilityHint: "TIMESTAMP to NAME casts are dependent on DateStyle; consider " +
 				"using to_char(timestamp) instead.",
 			dateStyleAffected: true,
 		},
 		oid.T_text: {
 			maxContext: CastContextAssignment,
 			origin:     contextOriginAutomaticIOConversion,
-			volatility: volatilityTODO,
-			volatilityHint: "TIMESTAMP to TEXT casts are dependent on DateStyle; consider " +
+			// TODO(mgartner): This should be stable.
```

Comment:
you could probably ignore this TODO with enough time as @e-mbrown should be making these stable soon

Reply:
I'll leave here for posterity and @e-mbrown can remove it when the change to stable is made.